src/megaapi_impl.cpp

```diff
@@ -20649,6 +20670,38 @@ void MegaApiImpl::sendPendingRequests()
 
             if (!flag)
             {
+                long long cancelledPending = 0;
+                // 1. cancel queued transfers, not yet started (and not added to cache), up to the last one queued when cancelTransfers was queued
+                auto lastQueuedTransfer = request->getTransferTag();
+                while (MegaTransferPrivate *transfer = transferQueue.pop())
+                {
+                    if (transfer->getPlaceInQueue() > lastQueuedTransfer)
+                    {
+                        break;
+                    }
+                    if (!transfer->isSyncTransfer() && transfer->getType() == direction)
+                    {
+                        fireOnTransferStart(transfer);
+                        transfer->setState(MegaTransfer::STATE_CANCELLED);
+                        fireOnTransferFinish(transfer, make_unique<MegaErrorPrivate>(API_EINCOMPLETE), committer);
```

Comment:
I wonder if it would be a better user experience if we were to introduce a new error code EUSERCANCELLED which would reassure people that the transfer cancelled due to their own action, and not some other cause?  If course we would need to show that in the GUI too

Reply:
I believe this might go beyond the scope of this PR, don't you think? :relaxed:

Reply:
Yes - just a thought for the future - actually since we set state cancelled, maybe the GUI can use that to indicate to the user that the transfer stopped due to their own action, rather than being something they need to worry about - perhaps we don't need the error code.

Reply:
Yes, as a matter of fact, MEGAsync does not show CANCELLED transfers AFAIK